clusterloader2/pkg/measurement/measurement_executor.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package measurement
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/perf-tests/clusterloader2/api"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+)
+
+// Execute executes a measurement, which can be a single measurement or a wrapper for multiple measurements.
+// It throws an error if both identifier and Instances has been supplied in the config.
+func Execute(mm MeasurementManager, m *api.Measurement) error {
+	if len(m.Instances) != 0 && m.Identifier != "" {
+		return fmt.Errorf("only one of instances or identifier must be supplied. Measurement method - %s, identifier - %s, instances - %v", m.Identifier, m.Method, m.Instances)
+	}
+	if m.Identifier != "" {
+		return mm.Execute(m.Method, m.Identifier, m.Params)
+	}
+	return executeWrapperMeasurement(mm, m)
+}
+
+func executeWrapperMeasurement(mm MeasurementManager, m *api.Measurement) error {
+	var wg wait.Group
+	errList := errors.NewErrorList()
+	for i := range m.Instances {
+		identifier := m.Instances[i].Identifier
+		measurementInstanceParams := m.Instances[i].Params
+		measurementParams := make(map[string]interface{})
+		for k, v := range m.Params {
+			measurementParams[k] = v
+		}
+		if measurementInstanceParams == nil {
+			measurementInstanceParams = measurementParams
```

Comment:
NIT: Do we need this if clause? Wouldn't be enough to have
```
for paramKey := range measurementInstanceParams {
...
}
```

I think a nil map is treated as empty map when ranging over it.

Reply:
So we can just have
```
measurementParams := make(map[string]interface{})
for k, v := range m.Params {
  measurementParams[k] = v
}
for k, v := range m.Instances[i].Params {
  measurementParams[k] = v
}
```

Reply:
yeah i messed it again. corrected it